### PR TITLE
Clarify the Purpose in the context/narrative

### PR DESCRIPTION
**Pull request by chexxor**
I wonder if this change/clarification of the Purpose is correct, appropriate, and would help clarify how the rest of this document should be written. Let me know what you think when you have time @JordanMartinez.

This is motivated by my comment from https://github.com/chexxor/purescript-documentation-discussion/pull/50#issuecomment-475916472

### CODE REVIEW

**Comment by chexxor** [comment]
I wonder if the Purpose and the ToC should have a more direct mapping.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I think I'm against the new rendering. The second sentence is a fragment, not a sentence. Perhaps that's a good thing for skimmability?

Still, I like how you moved the 'size of change can decrease its usefulness' down into its related section. Seems like a better spot to put it. Maybe it should just have the top sentence and nothing more? If so, it might be rendered as:
> Third, documentation becomes outdated/inaccurate due to changes, especially breaking changes, in two ways (see below).

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: This makes it sound like the audience is only for PS learners. Isn't this also for PS users and core contributors?

If we say 'either want to learn how to use PS or already believe in using it instead of alternatives,' I'd be fine with that.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: We're using a different TOC generator now? Is it better than the previous one in some way?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: "Determine" rather than "Provide" seems to be more of an investigation rather than a "Hey, there's fire in that room over there, so here's how to navigate through it." I also think the 'seen as lacking' is a bit misleading. People already believe it is lacking. That rendering makes it sound like we're going to argue why it's not actually lacking.

As for the second point's rendering, perhaps it could use "Summarize" rather than "understand." The latter makes it sound like we're going to ask questions rather than say, "Here's the gist of things." I like the 'expectations and needs' part. 
I think the 'documentation consumers' part is a bit narrow because of the 'consumer' part. It seems like it's only addressed to new learners (who consume docs) rather than a wider audience (some of who may write/produce docs and others who affect the documentation situation).

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I think this is fine (meaning, fine to include some of our ideas for how things could improve). It seems to be the direction we keep naturally heading towards.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I like that this will have a ToC section. However, I wonder whether this should go after the 'why read this document' section. Otherwise, that section isn't the first thing people see and they might think it's just going to be a massive ToC before the get to the real content, so they leave the page and go elsewhere.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:24]: Does this imply we're dropping the 'ps doc writers' audience entirely? Or that we aren't covering it specifically in this particular file but will in a separate file?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Thanks for your work and edits. I have a number of questions the rendering in some areas and feedback in others.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Yeah, I agree with that.

What do you think about this?

Rather than

"...due to changes"

it says

"...due to changes in the subject of the documentation..."

I'd prefer the latter because, as I understand it, the problem is the documentation becoming out-of-sync with the thing being documented, but the former way of saying it makes me think the problem is changes in the documentation itself. Am I misunderstanding that?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: 'We assume that readers are already interested in PureScript'. I think this would generalize the intended meaning well enough. What do you think?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I actually couldn't find the link to the other one I used. 😅 Thanks for raising this point. I'll find the other one I used.

Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Alright, I found the other TOC generator and used that one. I spent like 10 minutes trying to find it from a Google search last time, but I quickly found it this time when searching "TOC" in this project's Github search.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: > I'd prefer the latter because, as I understand it, the problem is the documentation becoming out-of-sync with the thing being documented, but the former way of saying it makes me think the problem is changes in the documentation itself. Am I misunderstanding that?

Ah... I see your point now. That's a good point.
I think what trips me up then is seeing 'documentation' appear twice in the same sentence. Perhaps it could be changed to 'documentation becomes inaccurate/outdated when the thing/subject being documented changes' 
Or 'documentation becomes inaccurate/outdated when changes occur in the thing being documented.'  ?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Mm... that sounds better. Still, I think it could be broadened slightly to include more of the notion of users besides those who are just interested. How about this rendering?
> We assume that readers are already interested in or use PureScript.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I thought I linked to it in #57 :smile:

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: Good ideas, I'll change it to that point of view.

The reason I wanted to add some qualification around "lacking" is because I believe that's fundamentally an opinionated thing. A thing is always seen as lacking by someone, even if 99% of people think its perfect. Even within the group of people who believe it is perfect, they are likely to change their mind when they are having problems understanding how to write an app which has a unique problem that doesn't have documentation or guidance because it is so unique.

So, I think "lacking" is unavoidable, but it *is* feasible to bring the ratio of people who feel its lacking to a more healthy number. For example, 

    - New learners - 10% of audience is satisfied
    - PureScript documentation writers - 50% of audience is satisfied
    - Core contributors - 60% of audience is satisfied

If these figures are current ratios of the satisfied consumers of the documentation, then perhaps a good goal is 75% or 85% for each of those audiences.

Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: I added a commit which updates the Purpose section. I don't know if it correctly addresses your concerns, but I think it has a more clear break-down. I think it needs another review - I'll certainly review it again later myself.

Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: > The reason I wanted to add some qualification around "lacking" is because I believe that's fundamentally an opinionated thing. A thing is always seen as lacking by someone, even if 99% of people think its perfect.

Good points! I assume we'd change `Determine` to `Explain` then? Or how would you render it now? 
Right now (with the 'explain' word used instead of 'determine'), I read this as "we'll be explaining what each relevant party (new learners, users, core contributors) thinks of the situation" rather than "we'll be explaining what are the issues behind why it's bad, regardless of who thinks that." (AFAIK, every relevant party thinks its bad)

> If these figures are current ratios of the satisfied consumers of the documentation, then perhaps a good goal is 75% or 85% for each of those audiences.

That sounds like a good goal. How would it be quantified? What does a 100% for new learners mean or look like? How is it different from a 100% for a core contributor?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I like that idea. I removed the "## Why Read This Document?" section because, in my experience, the TOC is usually in the very top section to introduce a navigation to and overview of all following sections. LMK what you think about this change.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:24]: No, it's in the TOC as "PureScript Documentation Writers: What is the Best Way to Write Documentation in this Context?". I intended the TOC to replace this "Outline", as that's one of the main functions of a TOC, I think.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I'm restating what I already stated to make sure my prior comment doesn't get overlooked.

I think this should be rendered as "we assume that readers are already interested in or use it.' The 'already interested in' refers to new learners. The 'use' refer to users and core contributors.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Yeah, I think that would be more clear, for sure. I have a habit of using "proper" grammar, often going so far as to make smooth reading more difficult, regrettably. I'll change that now.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: I was thinking "interested people" includes users and contributors, actually. If they weren't interested in it, they wouldn't be using or contributing to it, right? But yeah, it's true that people could be coerced into using PureScript, such as for a school project or for work, so yeah your phrasing is probably better.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: As long as we keep the 'why read this document' section brief, I think both concerns are still adequately addressed (the navigational "norm" is still a principle to which we adhere, and people have small enough thing to read that they do not immediately navigate away when they shouldn't)

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:24]: Ah, gotcha!

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Yeah.... it's frustrating isn't it? Why can't people just read properly?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:116]: These spelling errors, man.... :man_facepalming:

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [inline 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Should this be turned into a bullet list to make it more skimmable / make the sentence less long?

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: I imagine a 100% score would mean that a person from that audience will find an answer to their PureScript-related question 100% of the time. Another way of saying this, perhaps, is that documentation contains answers for 100% of the PureScript-related questions a PureScript audience has.

So, to think a bit further, a 20% score means that the documentations answers 20% of the questions a person has, regardless of how common those questions are. I guess if 1 really common problem is documented well, it would raise the score equally as much as documenting 10 niche questions well would raise the score.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Yeah, good idea. I'll do that now.

Relatedly, I read some of the following sections which explain these two points, and it seems I was wrong when thinking this section is about "the thing being documented", mainly because the second point, the frequency of changes", seems to be talking about the documentation rather than the thing being documented.

Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md]: Oh. No, I meant the thing being documented when talking about the frequency of changes, not the documentation itself. So, I believe you read that correctly.

As an example, if a library was released as v1.0.0 and then later patched (v1.0.2) and then bumped (v1.1.0) and then patched again (v1.1.1) all within a short time period, that outdates the documentation because it doesn't fully cover the library. However, the 'outdatedness' of the docs is not the same as what occurs in breaking changes.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: I wonder if it's worth finding a place in this project to document this idea of a "documentation score". Having metrics can be a double-edged sword: it can increase focus, but if a metric is inappropriately defined or measures the wrong thing then the project moves in a slightly wrong direction and doesn't serve its audience as effectively as it could. (I hear that often the bad side of metrics is people gaming them for personal gain, but I suspect that wouldn't be a problem in this situation, so I say 'moves it in a slightly wrong direction'.)

Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: Good points. It sounds like you're describing something like this:
- Findable: person can find answers to their questions (in ideally 5 minutes or less of Googling)
- Coverage: most common questions are well answered (niche ones might not be worth it as much, or just direct people to ask on the Slack chatroom)

--wait, isn't this just redescribing what we've already learned from the 'how to write good docs' section?

Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: > Having metrics can be a double-edged sword

Very true. They're useful when they're useful and painful in practically all other situations.

Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: I don't recall seeing this exact thing in the stuff I read, but it's likely that I overlooked it. :)

Comment (chexxor) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: Also yeah, it's probably best to keep the metric idea unmeasured for now. :)

**Comment by chexxor** [comment]
I'll revisit this comment after merging this: https://github.com/chexxor/purescript-documentation-discussion/pull/60#discussion_r270611559

I think all your comments can be reviewed by you again and marked as resolved. Following that, I'll merge this.

File: 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md
Comment (JordanMartinez) [reply 02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md:11]: > I don't recall seeing this exact thing in the stuff I read, but it's likely that I overlooked it. 

I don't think it is, but one point (docs should be easily navigatable) implies things like the above

**Comment by JordanMartinez** [comment]
Let's just merge this.

As a reminder to myself, here's the things I want to reconsider in the Context PR (#50):
- [my comment about the wording of the purpose of the document](https://github.com/chexxor/purescript-documentation-discussion/pull/60/files#r270259287)
- [ensure we keep the starting summary short and list the ToC immediately following it](https://github.com/chexxor/purescript-documentation-discussion/pull/60#discussion_r270611228)
- [further clarify possible factors that can be used to determine whether docs are good or not (but don't quantify them)](https://github.com/chexxor/purescript-documentation-discussion/pull/60#discussion_r270611663)

**Review by JordanMartinez — APPROVED** [review]

**Pull request merged**
